pinot-spi/src/main/java/org/apache/pinot/spi/stream/OffsetCriteria.java

```diff
@@ -168,24 +183,30 @@ public OffsetCriteria withOffsetCustom(String customString) {
      */
     public OffsetCriteria withOffsetString(String offsetString) {
       Preconditions.checkNotNull(offsetString, "Must provide offset string");
+      _offsetCriteria.setOffsetString(offsetString);
 
       if (offsetString.equalsIgnoreCase(OffsetType.SMALLEST.toString())) {
         _offsetCriteria.setOffsetType(OffsetType.SMALLEST);
       } else if (offsetString.equalsIgnoreCase(OffsetType.LARGEST.toString())) {
         _offsetCriteria.setOffsetType(OffsetType.LARGEST);
       } else {
-        try {
-          Long periodToMillis = TimeUtils.convertPeriodToMillis(offsetString);
-          if (periodToMillis >= 0) {
-            _offsetCriteria.setOffsetType(OffsetType.PERIOD);
+        Long periodToMillis = TimeUtils.convertPeriodToMillis(offsetString);
+        if (periodToMillis >= 0) {
+          _offsetCriteria.setOffsetType(OffsetType.PERIOD);
+          return _offsetCriteria;
+        } else {
+          LOGGER.error("Invalid time spec: '" + offsetString + "' (Valid examples: '3h', '4h30m')");
+          Long timestampToMillis = TimeUtils.convertDateTimeStringToMillis(offsetString);
```

Comment:
This is problematic. If offsetString is not of type period, convertPeriodToMillis throws parsing exception and it doesn't get to the else part.